### DISCUSSION TITLE
Plan to use Rust language?

### DISCUSSION THREAD

**Discussion by chojpsh1** [zephyrproject-rtos/zephyr#52871] [Q&A] [answered]
Hello, I wonder if there is plan to use the Rust in Zephyr kernel.
I would appreciate if let me know any plans or contents already mentioned about Rust in Zephyr (I tried to find it,but I couldn't)
Thank you!

**Comment by marc-hb** [comment]
I don't know about any "plan" but you may find this interesting: https://zephyr2022.sched.com/event/10CRS/using-zephyr-with-non-c-languages-david-brown-linaro https://www.youtube.com/watch?v=4cZK6ceNMX0

**Reply by chojpsh1** [reply]
Rust is mentioned in there. Thank you for finding it. What I'm curious about is a little solved. Thank you!

**Comment by arbrauns** [comment]
I don't think this would make all that much sense. Zephyr is a small enough OS that writing something equivalent from scratch in another language is feasible; that would avoid all the "worst of both worlds"-style tradeoffs that always come with mixing languages. For example, zephyr uses preprocessor macros extensively, unless you want to send your Rust code through the C preprocessor (probably a bad idea), you would have to recreate the entire devicetree macro system and maintain two implementations side-by-side.

**Reply by chojpsh1** [reply]
I see. Thank you for your reply!

**Comment by fkromer** [comment, marked as answer]
The answer is outdated. There is an official effort going on to support Rust: https://github.com/zephyrproject-rtos/zephyr-lang-rust For the current state of progress: https://eoss24.sched.com/event/1aBFG/update-on-rust-in-zephyr-david-brown-linaro-ltd
